src/android/BluetoothLePlugin.java

```diff
@@ -2905,6 +2905,9 @@ public void onScanResult(int callbackType, ScanResult result) {
           JSONObject returnObj = new JSONObject();
 
           addDevice(returnObj, result.getDevice());
+          if(result.getScanRecord().getDeviceName() != null){
+            addProperty(returnObj, keyName, result.getScanRecord().getDeviceName().replace("\0", ""));
```

Comment:
Ahh I see what you're saying. The empty stay is just right padded with 0x00? I think that behavior is fine then.